core/src/main/scala/org/broadinstitute/dsde/rawls/snapshot/WrappedSnapshot.scala

```diff
@@ -0,0 +1,34 @@
+package org.broadinstitute.dsde.rawls.snapshot
+
+import bio.terra.datarepo.model.{CloudPlatform, DatasetSummaryModel, SnapshotModel}
+import org.broadinstitute.dsde.rawls.RawlsException
+
+import scala.jdk.CollectionConverters.CollectionHasAsScala
+
+// A basic wrapper class to simplify interaction with the SnapshotModel from TDR
+class WrappedSnapshot(snapshot: SnapshotModel) {
+
+  // While getSource returns a list of SnapshotSourceModels, this is an artifact and we can make an assumption that
+  // a given snapshot will have exactly one source that will always have a dataset.  Violations of this assumption
+  // indicate a bug, in which case, crash and burn.
+  private def snapshotDataset(): DatasetSummaryModel = getOnlyElement(snapshot.getSource.asScala.toList).getDataset
+
+  private def getOnlyElement[A](list: List[A]): A = list match {
+    case onlyElement :: Nil => onlyElement
+    case _                  => throw new NoSuchElementException(s"Expected exactly one element, but found ${list.size}")
+  }
+
+  def isProtected: Boolean = Option(snapshotDataset().isSecureMonitoringEnabled)
+    .getOrElse(
+      throw new RawlsException(
```

Comment:
Any way to get coverage on this exception handling?

Reply:
@aherbst-broad @davidangb added one more commit with the additional test coverage requested

You'll note it includes an `ignore`d spec with the _actual, desirable_ behavior, and a copout 'working as implemented' spec that clarifies current behavior, with great disappointment.  I kind of want to roll forward with this anyway because I don't think I'm _introducing_ the described problem, just surfacing it.

Final thoughts, in the form of thumbs up or down, would be welcome.

Reply:
Discussed this with @davidangb and he helped me realize I was just missing a key part of the fixture, which was to set the boolean explicitly to `null` (otherwise it'd use the default value of the instance initializers).